packit/config/package_config.py

```diff
@@ -254,6 +254,151 @@ def __eq__(self, other: object):
         )
 
 
+class PackageConfigSchemaValidation:
+    @staticmethod
+    def validate_primitive(package_config: PackageConfig) -> dict:
+        from packit.schema import PackageConfigSchema
+
+        return PackageConfigSchema().validate(
+            {
+                "config_file_path": package_config.config_file_path,
+                "specfile_path": package_config.specfile_path,
+                "downstream_package_name": package_config.downstream_package_name,
+                "upstream_project_url": package_config.upstream_project_url,
+                "upstream_package_name": package_config.upstream_package_name,
+                "upstream_ref": package_config.upstream_ref,
+                "upstream_tag_template": package_config.upstream_tag_template,
+                "dist_git_base_url": package_config.dist_git_base_url,
+                "dist_git_namespace": package_config.dist_git_namespace,
+                "spec_source_id": package_config.spec_source_id,
+                "create_pr": package_config.create_pr,
+                "actions": package_config.actions,
+            }
+        )
+
+    @staticmethod
+    def validate_list(package_config: PackageConfig) -> dict:
+        from packit.schema import PackageConfigSchema
+
+        return PackageConfigSchema().validate(
+            {
+                "create_tarball_command": package_config.create_tarball_command,
+                "current_version_command": package_config.current_version_command,
+                "allowed_gpg_keys": package_config.allowed_gpg_keys,
+                "patch_generation_ignore_paths": package_config.patch_generation_ignore_paths,
+            }
+        )
+
+
+class PackageConfigValidation:
+    def __init__(self, package_config: PackageConfig):
+        self.config_file_path: Optional[str] = package_config.config_file_path
+        self.schema_primitive_errors: str = str(
+            PrimitiveValidationError(package_config)
+        )
+        self.schema_list_errors: str = str(ListValidationError(package_config))
+
+    def get_validation_errors_str(self) -> str:
+        if not (self.schema_primitive_errors or self.schema_list_errors):
+            return f"{self.config_file_path} is valid and ready to be used"
+
+        validation_errors = f"{self.config_file_path} does not pass validation:\n"
+        validation_errors += self.schema_primitive_errors
+        validation_errors += self.schema_list_errors
+
+        return validation_errors
+
+    def __repr__(self) -> str:
+        return self.get_validation_errors_str()
```

Comment:
I would maybe differentiate between `__repr__`/`__str__` and output used for users.

- `__str__`/`__repr__` something like `PackageConfigValidation(config_file_path='the/path/to/config.yaml)` or `PackageConfigValidation(config_file_path='the/path/to/config.yaml, errors=['error1', 'error2'])`
  - Developer-focused view, I would not use those for forwarding user output. Somer property for accessing errors should be enough and the formatting can be done elsewhere.
- output printed to the user:
```
We found the following errors:
- error 1
- error 2
```
  - What about implementing that in the CLI method and not here? (All used attributes can be used from outside of the class as well.) It allows us easily update/tweak/colour the output.


Same can be applied to other classes.